core/include/traccc/edm/impl/silicon_cell_collection.ipp

```diff
@@ -0,0 +1,66 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022-2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Local include(s).
+#include "traccc/definitions/math.hpp"
+
+namespace traccc::edm {
+
+template <typename BASE>
+TRACCC_HOST_DEVICE silicon_cell<BASE>& silicon_cell<BASE>::operator=(
+    const silicon_cell& other) {
+
+    channel0() = other.channel0();
+    channel1() = other.channel1();
+    activation() = other.activation();
+    time() = other.time();
+    module_index() = other.module_index();
+    return *this;
+}
+
+template <typename BASE>
+template <typename T, std::enable_if_t<!std::is_same_v<BASE, T>, bool> >
+TRACCC_HOST_DEVICE silicon_cell<BASE>& silicon_cell<BASE>::operator=(
+    const silicon_cell<T>& other) {
+
+    channel0() = other.channel0();
+    channel1() = other.channel1();
+    activation() = other.activation();
+    time() = other.time();
+    module_index() = other.module_index();
+    return *this;
+}
+
+template <typename BASE>
+template <typename T>
+TRACCC_HOST_DEVICE bool silicon_cell<BASE>::operator==(
+    const silicon_cell<T>& other) const {
+
+    return (channel0() == other.channel0()) &&
+           (channel1() == other.channel1()) &&
+           (math::fabs(activation() - other.activation()) < 1e-6f) &&
+           (math::fabs(time() - other.time()) < 1e-6f) &&
+           (module_index() == other.module_index());
```

Comment:
```suggestion
    return operator<=>(other) == std::weak_ordering::equivalent;
```

Reply:
For the ordering, we are fine with weak ordering. Our algorithms don't care whether cells coming from different truth particles are in one order during reconstruction or in some other.

However, the equality operator is used in our efficiency measurement code extensively. Where we do care about which cell belongs to which truth particle. So for the equality comparison we need strong ordering. But I didn't want to slow down our reconstruction code with strong ordering.

Long story short, I believe the current code is the way to go.

Reply:
Okay this is actually a fair point. :+1: